htsget.md

```diff
@@ -357,15 +357,119 @@ Initial guidelines, which we expect to revise in light of future experience:
+<tr markdown="block"><td>`fields`  _optional array of strings_</td><td>
+List of fields to include.  See Field filtering.
+  <table>
+     <tr markdown="block"><td>Field name _string_</td></tr>
+  </table>
+</td></tr>
+<tr markdown="block"><td>`tags`  _optional array of strings_</td><td>
+List of tags to include, default: all.  If an empty array is specified, no tags are included.
+
+The server SHOULD respond with an `InvalidInput` error if `tags` and `notags` intersect.
+  <table>
+     <tr markdown="block"><td>Tag name _string_</td></tr>
+  </table>
+</td></tr>
+<tr markdown="block"><td>`notags`  _optional array of strings_</td><td>
+List of tags to exclude, default: none.  If an empty array is specified, no tags are included.
+
+The server SHOULD respond with an `InvalidInput` error if `tags` and `notags` intersect.
+  <table>
+     <tr markdown="block"><td>Tag name _string_</td></tr>
+  </table>
+</td></tr>
+<tr markdown="block"><td>`regions` _object_</td><td>
+Regions to return.  If not present, the entire file will be returned.  If an empty object or null, just the header will be returned.
+
+Each name/value pair in the object defines a reference name and the list of
+locations on that reference to return.
+
+The server SHOULD respond with an `InvalidInput` error if the region list is not well-formed.
+  <table>
+     <tr markdown="block"><td>referenceName _string_</td>
+        <td>Locations _array of two-element arrays_
+           <table>
+              <tr markdown="block"><td>
+                 <table>
+                    <tr markdown="block"><td>start _number_</td>
+                      <td>end _number or string "end"_</td></tr>
+                  </table>
+              </td></tr>
+           </table>
+        </td></tr>
+  </table>
+</td></tr>
+</table>
+
+### Example
+
+```json
+{
+   "format" : "bam",
+   "fields" : ["QNAME", "FLAG", "RNAME", "POS", "CIGAR", "SEQ"],
+   "tags" : ["RG"],
+   "notags" : ["OQ"],
+   "regions" : {
+      "chr1" : [[0, "end"]],
+      "chr2" : [[1000, 1001], [2000, 2100]]
+   }
+}
+```
+
+## Region list
+
+The `regions` parameter is an object with a key for each reference to be returned and a value which lists the desired locations on that reference.
+The location list is an array of two-element `[start, end]` arrays.
+Both elements must always be present.
+`start` must always be a number and gives the 0-based, inclusive start position.
+`end` must either be a number giving the 0-based end position, or the exact string `"end"` to indicate that the region finished at the last base of the given reference.
+The end position must be strictly greater than the start.
+To return all bases for `chr1`, the client should use the request
+`"regions" : { "chr1" : [[0, "end"]]}`.
+To return a single base, the client should use `[POS, POS+1]`, for example:
+`"regions" : { "chr1" : [[1000, 1001]]}`.
+
+If elements in the location list overlap, the returned data will be the union of the overlapping elements.
```

Comment:
Does this make life hard for the server unnecessarily? You could imagine that if a client asks for the same region twice, then it's perfectly fine to just return the same data twice. I don't really know, just asking the question!

Reply:
I thought it would be best to define what happens in this case so we don't end up with a mixture of servers that repeat the data and ones that don't.  The server would have to do similar work anyway so that it can handle non-overlapping locations that have lots of reads in common because they are close enough together.

Reply:
Definitely agree that we should define the behaviour, and I agree with your argument. The reason I bring up the alternative, is that the obvious and easy way to implement this from a server perspective (in terms of code, not necessarily efficiency) is to just treat this in this as a list of independent GET requests, and just concatenate the lists of chunks. The "front-end" server can't do this if it has to reason about overlaps within the requests. 

I agree that if minimising the traffic to the back-end server is the goal, then returning the union is the way to go.